sdks/java/io/iceberg/src/test/java/org/apache/beam/sdk/io/iceberg/RecordWriterManagerTest.java

```diff
@@ -465,6 +486,10 @@ public void testIdentityPartitioning() throws IOException {
             .identity("float")
             .identity("double")
             .identity("str")
+            .identity("date")
```

Comment:
for my knowledge, it's unclear to me how `testIdentityPartitioning()` was passing [when identity should support Any source type](https://iceberg.apache.org/spec/#partition-transforms). Is it worth adding an integration test to ensure just the `Write` works when partitioning by just `date`, as well as `Any`?

Reply:
This test used to not include `date` and the other time types. This PR adds adds them to the test (your comment is actually positioned on the lines that add them).